### ISSUE TITLE
Submit start with a slash

### ISSUE THREAD

**Issue by moueza** [zowe/vscode-extension-for-zowe#1116]
From my answer https://github.com/zowe/jobs/issues/38 :

Reproduced in openmainframeproject : on VS code Linux Ubuntu 20.10 cbl0002 (6. of 10.4 Lab of pdf1) : right click Submit job :
`Job submission failed
z/OSMF REST API Error:
Rest API failure with HTTP(S) status 400
rc: 4
reason: 17
category: 6
message: Submit input data does not start with a slash

Error: z/OSMF REST API Error:
Rest API failure with HTTP(S) status 400
rc: 4
reason: 17
category: 6
message: Submit input data does not start with a slash

`

**Comment by Marjona6** [comment]
Are you submitting the COBOL source code or the JCL? I had this error when trying to submit the COBOL source.

**Comment by moueza** [comment]
I had no JCL ( because I had not filtered for id.*) ... so I did from CBL.

You must do from JCL (by well filtering), and it is OK !

SOLVED

**Comment by sravanisf8** [comment]
Command Error:
z/OSMF REST API Error:
Rest API failure with HTTP(S) status 400
rc:       4
reason:   17
category: 6
message:  Submit input data does not start with a slash

Error Details:
HTTP(S) error status "400" received.
Review request details (resource, base path, credentials, payload) and ensure correctness.

  
Can someone help me how to overcome the above error in ZOWE

**Comment by zFernand0** [comment]
@LukeG-96 
Seems like this was related to an operation (filtering) done on the JCL before being submitted that caused the issue.
Could you verify that the JCL being submitted is properly formatted?

**Comment by LukeG-96** [comment]
@zFernand0 
So what's weird is when I right click on the member name and click "Pull from mainframe" it adds 1 space in front of each line of the JCL. So I delete that line, save the member and try to re-run the job and it still gives me this same error. I am not sure what is causing this reformat to happen, but it basically makes the option to use zowe through vs code unusable...

**Comment by zFernand0** [comment]
**TL;DR**: This could be related to a race condition in how we save and upload contents to the mainframe
- Details: https://github.com/zowe/vscode-extension-for-zowe/pull/2164

Workaround:
If you get the old contents (i.e. 1 space added in front of each line) after doing `Pull From Mainframe`, please make sure that the content is successfully uploaded after removing the extra space in front.

A quick way I personally work around this race condition is by triggering the upload request multiple times. For example:
1. adding a new line at the end of the file, 
2. saving the file, 
3. removing the line, and 
4. saving the file.

After doing these steps, I do a `Pull From Mainframe` and the extra space in front of each line does not show up anymore 😋 

@LukeG-96, Please let me know if that is the case for you as well 😋

**Comment by LukeG-96** [comment]
I get the "Data Set Uploaded successfully" message but when I follow the steps you suggested it still puts one space in front of every line and adds a new line at the bottom...

Below is what is looks like after I make my changes and save the member:
![image](https://user-images.githubusercontent.com/126080946/222794372-76ea0718-9f6a-4088-b596-2c923f1756bb.png)

Here is what it looks like after I issue a **Pull from mainframe**
![image](https://user-images.githubusercontent.com/126080946/222794469-4cc8cebd-e7db-461b-ac97-005f10023680.png)

**Comment by zFernand0** [comment]
Moving this conversation to a new issue. 
Thanks for taking the time to provide more details 👍
